File: benchmark/benchmark.py
import math
import time

def estimate_pi(n):
    s = 1.0
    for i in range(1, n + 1):
        if i % 2 == 1:
            si = -1 / (2 * i + 1)
        else:
            si = 1 / (2 * i + 1)
        s += si
    return 4 * s

t0 = time.time()
my_pi = estimate_pi(100_000_000)
t1 = time.time()
total = t1 - t0
print("Estimate of pi: ", my_pi)
print("Error from exact value: ", math.pi - my_pi)
print("Time: ", total, " seconds")

File: benchmark/benchmark.cpp
#include <iostream>

double estimate_pi(int n) {
  double s = 1.0;
  double si;
  for(int i = 1; i < n; i++) {
    if(i % 2 == 1) {
      si = -1.0 / (2.0 * i + 1.0);
    }
    else {
      si = 1.0 / (2.0 * i + 1.0);
    }
    s += si;
  }
  return 4 * s;
}

int main()
{
  auto s = estimate_pi(100000000);
  //std::cout << s << std::endl;
  return 0;
}